src/dbnode/storage/index/convert/convert.go

```diff
@@ -211,7 +212,13 @@ func TagsFromTagsIter(
 			idRef = true
 		} else {
 			if idPool != nil {
-				tag.Name = idPool.Clone(curr.Name)
+				// NB(r): Fast path for if a graphite tag name to save
+				// a lot of space is to reuse a preallocated tag name.
+				if idx, ok := graphite.TagIndex(nameBytes); ok {
```

Comment:
Might worth either put this change in a separate PR or call it out. this is a very nice change with decent cpu reduction.